src/plugins/saved_objects_management/public/management_section/objects_table/components/table.tsx

```diff
@@ -74,6 +74,8 @@ interface TableState {
   activeAction?: SavedObjectsManagementAction;
 }
 
+const MAX_PAGINATED_ITEM = 10000;
```

Comment:
Not a big fan of this hardcoded limit, OTOH that's how it's done in other parts of Kibana too, and it's way easier than fetching the setting from all the SO indices targeted by the request.

Reply:
Can an end-user face the problem when requesting `/api/kibana/management/saved_objects/scroll/counts`? If so, maybe this logic belongs to the server-side?

Reply:
it can't when requesting `/api/kibana/management/saved_objects/scroll/counts`, or when performing the export, only when manually calling `/api/kibana/management/saved_objects/_find` if the requested page/perPage exceeds the threshold (but even if prefixed by `/api`, there shouldn't be a case where an integration uses this endpoint)